### DISCUSSION TITLE
Installation problem Ricoh-Aficio-SG-2100-n

### DISCUSSION THREAD

**Discussion by AR404lab** [OpenPrinting/cups#450] [Q&A]
Hello everyone, 
I have problems to install a Ricoh-Aficio-SG-2100-n printer on an Ubuntu 22.04.1 installation

I found some hints on German and Japanese forums but nothing conclusive. 

Do you have any clues so that I can install it?

**Comment by zdohnal** [comment]
Hi @AR404lab ,

it would be great if you said how you printer is connected. I've found a manual for your printer from 2011, so it can be possible you can printer driverlessly over network and a slim chance for driverless USB printing.

In case it is a network printer and you have enabled IPP port on the printer, try:

 $ ipptool -tv ipp://<printer_ip>/ipp/print get-printer-attributes.test

if this passes, your printer should be able to print without drivers.

In case it is connected by USB, check 'lsusb -v' output for 7/1/4 interface:

```
bInterfaceClass 7 Printer
bInterfaceSubClass 1 Printer
bInterfaceProtocol 4
```
if it has such interface, install ipp-usb.

In case you can use driverless printing and the printer is in local network or it is driverless USB, don't install the printer and use a temporary queue which will appear in any up-to-date print dialogs - you can check whether the temp queue is seen by ```lpstat -e```. If you find the temp queue under the previous output and your application doesn't show it in print dialog, report the bug to the application to support temporary queues.

In case you can use driverless printing and the printer is in different network, install the printer permanently by:

$ lpadmin -p <name> -v ipp://<printer_ip>/ipp/print -m everywhere -E

In case your printer can't do driverless printing, try some driver or printer application from similar printer model (check https://openprinting.org/printers/manufacturer/Ricoh ), or use a generic one (hopefully it will support pxlmono or postscript...).

And the last resort is installing the printer in Windows and share it via Samba to Ubuntu.

**Comment by AR404lab** [comment]
Hello, thank you for your message I tried the network printing here is the message I have 

> remy-ThinkPad-T420:~$ ipptool -tv ipp://192.168.1.26/ipp/print get-printer-attributes.test
"/usr/share/cups/ipptool/get-printer-attributes.test":
    Get-Printer-Attributes:
        attributes-charset (charset) = utf-8
        attributes-natural-language (naturalLanguage) = en
        printer-uri (uri) = ipp://192.168.1.26:631/ipp/print
        requested-attributes (1setOf keyword) = all,media-col-database
    Get printer attributes using get-printer-attributes                  [FAIL]
        RECEIVED: 33 bytes in response
        status-code = client-error-bad-request ("attributes-charset": Bad charset value "" - bad characters (RFC 8011 section 5.1.8).)
        attributes-charset (charset) = 
        Bad version 1.1 in response - expected 2.0 (RFC 8011 section 4.1.8).
        Bad request ID 0 in response - expected 52472 (RFC 8011 section 4.1.1)
        Missing second attribute "attributes-natural-language (naturalLanguage)" in group operation-attributes-tag (RFC 8011 section 4.1.4).
        "attributes-charset": Bad charset value "" - bad characters (RFC 8011 section 5.1.8).
        EXPECTED: STATUS successful-ok (got client-error-bad-request)
        EXPECTED: charset-configured
        EXPECTED: charset-supported
        EXPECTED: compression-supported
        EXPECTED: document-format-default
        EXPECTED: document-format-supported
        EXPECTED: generated-natural-language-supported
        EXPECTED: ipp-versions-supported
        EXPECTED: media-col-default
        EXPECTED: natural-language-configured
        EXPECTED: operations-supported
        EXPECTED: printer-info
        EXPECTED: printer-is-accepting-jobs
        EXPECTED: printer-location
        EXPECTED: printer-make-and-model
        EXPECTED: printer-more-info
        EXPECTED: printer-name
        EXPECTED: printer-state
        EXPECTED: printer-state-reasons
        EXPECTED: printer-up-time
        EXPECTED: printer-uri-supported
        EXPECTED: uri-authentication-supported
        EXPECTED: uri-security-supported

**Reply by zdohnal** [reply]
So the printer cannot work driverlessly - you can try drivers for similar printer models or generic drivers, or have the printer installed in Windows and share it via Samba.
